CMakeLists.txt

```diff
@@ -41,6 +41,15 @@ pkg_check_modules(MICROHTTPD REQUIRED libmicrohttpd)
 pkg_check_modules(SODIUM REQUIRED libsodium)
 find_package(Threads REQUIRED)
 
+if(NOT LINUX)
```

Comment:
This isn't a good way to do build conditionals. Need to check if epoll/argp work without extra libs, and check for extra libs if needed.

Noteworthy, some Linux distros do not have argp.